lib/local.js

```diff
@@ -1,10 +1,266 @@
+  const serviceDescription = `locally service "${serviceName}" of suite "${suiteName}" with coverage`
+  tracer.debug(`Starting ${serviceDescription}`)
+
+  const stdin = process.stdin
+  const coverageOptions = istabulMocha.istanbul.params(suite, suiteName).split(' ')
+  const commandAndParams = istabulMocha.getCommandAndParams(service.local.command)
+  let runServiceOptions = coverageOptions.concat(['cover'])
+    .concat([path.resolve(__dirname, 'service-coverage-runner.js')])
+    .concat(['--'])
+    .concat(commandAndParams.params.split(' '))
+  let proc
+
+  stdin.setRawMode(true)
+  stdin.resume()
+  stdin.setEncoding('utf8')
+
+  stdin.on('data', function (key) {
+    if (key === '\u0003') {
+      tracer.debug('CTRL-C received. Sending exit signal to service')
+      proc.send({ exit: true })
+    }
+  })
+
+  proc = childProcess.fork(path.resolve(__dirname, 'bin', 'msc-istanbul.js'), runServiceOptions, {
+    cwd: process.cwd(),
+    env: _.extend({}, process.env, {
+      servicePath: path.resolve(process.cwd(), commandAndParams.command)
+    })
+  })
+
+  proc.on('close', () => {
+    stdin.setRawMode(false)
+    stdin.pause()
+  })
+
+  return Promise.resolve({
+    process: proc,
+    name: serviceName,
+    isCoveraged: true
+  })
+}
+
+const logProcessData = function (data) {
+  data = _.trim(data)
+  if (data.length) {
+    console.log(data)
+  }
+}
+
+const runServiceCommand = function (suite, service, suiteName, serviceName) {
+  tracer.debug(`Starting locally service "${serviceName}" of suite "${suiteName}"`)
+  let proc = childProcess.execFile(service.local.command, [], {
+    cwd: process.cwd(),
+    env: _.extend({}, process.env, {
+      FORCE_COLOR: true
+    }),
+    shell: true,
+    windowsHide: true
+  })
+
+  proc.stdout.setEncoding('utf8')
+
+  proc.stdout.on('data', (data) => {
+    logProcessData(data)
+  })
+
+  proc.stderr.on('data', (data) => {
+    logProcessData(data)
+  })
+
+  return Promise.resolve({
+    process: proc,
+    name: serviceName
+  })
+}
+
+const runService = function (suite, suiteName, serviceName) {
+  return getServiceConfig(suite, suiteName, serviceName)
+    .then((service) => {
+      let errorMessage
+      if (!service.local || !service.local.command) {
+        errorMessage = `No local command found for service "${serviceName}" in suite "${suiteName}"`
+        tracer.error(errorMessage)
+        return Promise.reject(Boom.notFound(errorMessage))
+      }
+      if (suite.coverage && suite.coverage.from === serviceName && suite.coverage.enabled !== false) {
+        return runCoveragedService(suite, service, suiteName, serviceName)
+      }
+      return runServiceCommand(suite, service, suiteName, serviceName)
+    })
+}
+
+const resolveOnClose = function (localService) {
+  return new Promise((resolve, reject) => {
+    localService.process.on('close', (code) => {
+      tracer.debug(`Service "${localService.name}" closed with code ${code}`)
+      if (code && code !== 0) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Refactor this code so that this expression does not always evaluate to true. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=javascript%3AS2589)